src/Client/ClientBase.cpp

```diff
@@ -391,7 +391,7 @@ void ClientBase::initBlockOutputStream(const Block & block, ASTPtr parsed_query)
             output_format = global_context->getOutputFormat(
                 current_format, out_file_buf ? *out_file_buf : *out_buf, block);
 
-        output_format->doWritePrefix();
+        output_format->writePrefixIfNot();
```

Comment:
This is needed to avoid race in flushing data in ParallelFormattingOutputFormat (#31011). WIll be removed after fixing tgis problem

Reply:
It fixes `01249_flush_interactive`, but breaks test `01553_settings_early_apply`...

Reply:
Oh, got it. There is a problem on master:
```bash
$ clickhouse-client -q "select * from numbers(10) FORMAT TSVWithNamesAndTypes settings max_result_rows = 1"
number
UInt64
Received exception from server (version 21.12.1):
Code: 396. DB::Exception: Received from localhost:9000. DB::Exception: Limit for result exceeded, max rows: 1.00, current rows: 10.00. (TOO_MANY_ROWS_OR_BYTES)
(query: select * fromnumber
UInt64
Received exception from server (version 21.12.1):
Code: 396. DB::Exception: Received from localhost:9000. DB::Exception: Limit for result exceeded, max rows: 1.00, current rows: 10.00. (TOO_MANY_ROWS_OR_BYTES)
(query: select * from numbers(10) FORMAT TSVWithNamesAndTypes settings max_result_rows = 1) numbers(10) FORMAT TSVWithNamesAndTypes settings max_result_rows = 1)
```
It shouldn't print prefix in this case. My refactoring fixes it, but need to remove whis explicit `writePrefixIfNot` from client. Will wait for fixing #31011